src/python/bot/tasks/testcase_manager.py

```diff
@@ -526,121 +548,144 @@ def test_for_crash_with_retries(testcase,
           'No crash occurred (round {round_number}).'.format(
               round_number=round_number),
           output=output)
-      continue
 
+    return crash_result
+
+  def _pre_run_cleanup(self):
+    """Common cleanup before running a testcase."""
+    # Cleanup any existing application instances and user profile directories.
+    # Cleaning up temp clears user profile directories and should be done before
+    # calling |get_command_line_for_application| call since that creates
+    # dependencies in the profile folder.
+    process_handler.terminate_stale_application_instances()
+    shell.clear_temp_directory()
+
+  def _get_crash_state(self, round_number, crash_result):
+    """Get crash state from a CrashResult."""
     state = crash_result.get_symbolized_data()
     logs.log(
         ('Crash occurred in {crash_time} seconds (round {round_number}). '
          'State:\n{crash_state}').format(
-             crash_time=crash_time,
+             crash_time=crash_result.crash_time,
              round_number=round_number,
              crash_state=state.crash_state),
         output=state.crash_stacktrace)
 
-    if not compare_crash or not testcase.crash_state:
-      logs.log('Crash stacktrace comparison skipped.')
-      return crash_result
+    return state
 
-    if crash_result.should_ignore():
-      logs.log('Crash stacktrace matched ignore signatures, ignored.')
-      continue
+  def reproduce_with_retries(self,
+                             retries,
+                             expected_state=None,
+                             expected_security_flag=None,
+                             flaky_stacktrace=False):
+    """Try reproducing a crash with retries."""
+    self._pre_run_cleanup()
+    crash_result = None
 
-    if crash_result.is_security_issue() != testcase.security_flag:
-      logs.log('Crash security flag does not match, ignored.')
-      continue
+    for round_number in range(1, retries + 1):
+      crash_result = self.run(round_number)
+      state = self._get_crash_state(round_number, crash_result)
```

Comment:
can move this to line 604

Reply:
This logs the state even if we early exit in the following conditions, so we need this here.